### DISCUSSION TITLE
Error on upload file

### DISCUSSION THREAD

**Discussion by sotos89** [lookfirst/sardine#388] [Q&A]
Hello I am trying to upload file using webdav with java and getting this error 

`Cannot retry request with a non-repeatable request entity.` 

everything was working fine until some changes to the server, the changes are that the operating system is different and all configuration file are the same.

the access from the web page is working fine and the file downloading also work, the only problem is while uploading files.

I also tried to update to the latest version of the library and nothing changed.

any idea how to find more info about the error and to fix it?

**Comment by LotharSpiegel** [comment]
It is most likeley caused by the non-repeatable nature of an Inputstream. Could you post how you call the sardine put method? You could try to use the put method that accepts a byte array instead of an Inputstream and see if this already resolves the error. But be aware that this loads the complete content of the stream into the memory.

**Comment by pavelsalamon** [comment]
You have to enable pre-emptive auth or switch to byte[] entities which are repeatable, see comments in https://github.com/lookfirst/sardine/issues/103 for more info.
